src/Stream/MemoryStream.tpp

```diff
@@ -234,33 +200,26 @@ namespace Stream
 
       if (old_size > 0)
       {
-        Traits::copy(ptr, this->pbase(), old_size);
-        allocator_.deallocate(this->pbase(), old_size);
+        Traits::copy(ptr, begin_, old_size);
+        allocator_.deallocate(begin_, old_size);
       }
 
-      this->setp(ptr, ptr + new_size);
-      this->pbump(offset);
+      begin_ = ptr;
+      ptr_ = begin_ + offset;
+      end_ = begin_ + new_size;
+
       return true;
     }
 
     template <typename Elem, typename Traits, typename Allocator,
       typename AllocatorInitializer>
-    typename OutputMemoryBuffer<Elem, Traits, Allocator,
-      AllocatorInitializer>::Int
+    void
     OutputMemoryBuffer<Elem, Traits, Allocator, AllocatorInitializer>::
-      overflow(Int c) /*throw (eh::Exception)*/
+      incptr() throw()
```

Comment:
Why we need incptr ? Can we use pbump directly ?

Reply:
After additional review I think is more clear to move "append" into OutputMemoryBuffer and don't allow to manipulate with position directly (from public methods). And in this case you need only pbump

Reply:
as we discussed privately, we need pointer to memory region for to_chars.
also changed incptr to pbump(int) which is useful in append.